alb_no_logs.tf

```
@@ -54,18 +54,51 @@ resource "aws_lb_target_group" "main_no_logs" {
   }
 }
 
-resource "aws_lb_listener" "frontend_http_tcp_no_logs" {
+# resource "aws_lb_listener" "frontend_http_tcp_no_logs" {
+#   load_balancer_arn = "${element(concat(aws_lb.application_no_logs.*.arn, list("")), 0)}"
+#   port              = "${lookup(var.http_tcp_listeners[count.index], "port")}"
+#   protocol          = "${lookup(var.http_tcp_listeners[count.index], "protocol")}"
+#   count             = "${var.logging_enabled ? 0 : var.http_tcp_listeners_count}"
+
+#   default_action {
+#     target_group_arn = "${aws_lb_target_group.main_no_logs.*.id[lookup(var.http_tcp_listeners[count.index], "target_group_index", 0)]}"
+#     type             = "forward"
+#   }
+# }
+
+resource "aws_lb_listener" "frontend_http_tcp_forward_no_logs" {
   load_balancer_arn = "${element(concat(aws_lb.application_no_logs.*.arn, list("")), 0)}"
-  port              = "${lookup(var.http_tcp_listeners[count.index], "port")}"
-  protocol          = "${lookup(var.http_tcp_listeners[count.index], "protocol")}"
-  count             = "${var.logging_enabled ? 0 : var.http_tcp_listeners_count}"
+  port              = "${lookup(var.http_tcp_listeners_forward[count.index], "port")}"
+  protocol          = "${lookup(var.http_tcp_listeners_forward[count.index], "protocol")}"
+  count             = "${var.logging_enabled ? 0 : var.http_tcp_listeners_forward_count}"
 
   default_action {
-    target_group_arn = "${aws_lb_target_group.main_no_logs.*.id[lookup(var.http_tcp_listeners[count.index], "target_group_index", 0)]}"
+    target_group_arn = "${aws_lb_target_group.main_no_logs.*.id[lookup(var.http_tcp_listeners_forward[count.index], "target_group_index", 0)]}"
     type             = "forward"
   }
 }
 
+resource "aws_lb_listener" "frontend_http_tcp_redirect_no_logs" {
+  load_balancer_arn = "${element(concat(aws_lb.application_no_logs.*.arn, list("")), 0)}"
+  port              = "${lookup(var.http_tcp_listeners_redirect[count.index], "port")}"
+  protocol          = "${lookup(var.http_tcp_listeners_redirect[count.index], "protocol")}"
+  count             = "${var.logging_enabled ? 0 : var.http_tcp_listeners_redirect_count}"
+
+  default_action {
+    target_group_arn = "${aws_lb_target_group.main_no_logs.*.id[lookup(var.http_tcp_listeners_redirect[count.index], "target_group_index", 0)]}"
+    type             = "redirect"
+
+    redirect {
+      host        = "${lookup(var.http_tcp_listeners_redirect[count.index], "redirect_host", lookup(var.default_action_redirect_defaults, "host"))})}"
```

```suggestion
      host        = "${lookup(var.http_tcp_listeners_redirect[count.index], "redirect_host", lookup(var.default_action_redirect_defaults, "host"))}""
```

Here's a suggestion to fix it based https://github.com/terraform-aws-modules/terraform-aws-alb/commit/de6fe240dead819a1a9ce9d4fe9fdc7223ebe5f5 that's included in [my fork](https://github.com/arcadia/terraform-aws-alb).